proxy/proxy.go

```diff
@@ -254,11 +286,16 @@ func (c *filterContext) StateBag() map[string]interface{}    { return c.stateBag
 func (c *filterContext) BackendUrl() string                  { return c.backendUrl }
 func (c *filterContext) OriginalRequest() *http.Request      { return c.originalRequest }
 func (c *filterContext) OriginalResponse() *http.Response    { return c.originalResponse }
+
 func (c *filterContext) Serve(res *http.Response) {
 	c.servedWithResponse = true
 	c.res = res
 }
 
+func (c *filterContext) SetHost(h string) {
```

Comment:
no-no, just pushed now to discuss the preserveHost feature. cleanup and doc update is still missing, sorry. i'll make note when ready again for full review.